### PR TITLE
Merge feature branch to development

### PR DESCRIPTION
**Pull request by defoebrand**
This branch includes all relevant code to allow a user to create a new login account, create new events, and invite other users to their events. It does not allow users to edit other user's events. It displays events based on whether they are past or upcoming.

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @defoebrand and @m15e 
### You have done well so far :clap: However, you have a few issues to fix:-
- [ ] The views are not showing what they are supposed to show, for events it shows users
- [ ] No way a user can create events, invites friends
- [ ] Check the routing to much the views and action controllers
- [ ] Validate your models
### Make the changes and submit for another code review. Good luck.
- Happy hacking!

By Raymond
Slack @iamraymond

**Comment by m15e** [comment]
Dear second reviewer, we have added the validations to our models as per the previous TSE's request, however the issues 1-3 that are listed are unclear to us, as our README.md screenshot reflects that our events#index page shows what's required in the Odin project specification. Users can create events and invite guests in the same form and the routing works fine. If we're mistaken on this please advise otherwise.

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,11 @@
 Rails.application.routes.draw do
+  resources :events
+  resources :users
+  resources :sessions, only: %i[new create destroy]
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root 'events#index'
+
+  get 'signup', to: 'users#new', as: 'signup'
+  get 'login', to: 'sessions#new', as: 'login'
+  get 'logout', to: 'sessions#destroy', as: 'logout'
```
Comment (Dandush03) [inline config/routes.rb:10]: ### remember we required you to use `resources` try the following, check out the following suggestion :wink: 
```ruby
scope(path_names: { new: 'login', destroy: 'logout' }) do
  resources :session, path: ' '
end
```
>same goes for the `UserController`

**Review by Dandush03 — CHANGES_REQUESTED** [review]
# Status: :red_circle: Require Changes :red_circle: 
### Great Job! you're almost done
Remember that we favor no logic in our views, read this article -"[Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)"

also, check my comments on your routing.
# Happy Coding!

**Comment by m15e** [comment]
Dear TSEs, unfortunately the most recent review, like the one before that, is pointing out requests that are not very clear.

1.
The previous TSE is mentioning a requirement to use resources for our user/session routes, that is no where to be found on the [Odin Project Specifications](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations) for this project.

2. 
We understand that there should be minimal logic in the views with the exception of conditional show/hide functionality.

However, part of the project spec says to put logic in the view under Event Attendance (excerpt below)
"5. Modify the User’s Show page to separate those events which have occurred in the past (“Previously attended events”) from those which are occurring in the future (“Upcoming events”). You could do this by putting logic in your view. Don’t have your controller call separate model methods to retrieve each".

If this is a misunderstanding on our end, please advise.

Thank you

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
### Good job so far :+1: however, as mentioned by the previous reviewer; you need to move the logic from the view to the helper.

**_Check the Microverse instruction to [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/) under the "Additional Microverse Requirements" section to confirm this. If you check it you will see "Make sure that there is no logic in views and that helpers are responsible only for view related logic."_**

**_Kindly fix this and submit for another review._**

#### _Review by Temitayo Adeleke._

**Review by rOluochKe — APPROVED** [review]
### You finally got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy hacking!

**Pull request merged**